src/index.js

```diff
@@ -12,30 +12,40 @@ const defaultConfig = {
   "autoReconnectOnError": true
 };
 
-const client = new net.Socket();
-client.setEncoding('utf8');
-
 class Client {
-  submit(options) {
-    submitWork({
-      ...options,
-      client,
-    });
+
+  #client;
+
+  constructor(options) {
+    this.#start(options);
+  }
+
+  shutdown() {
+    this.#client.end();
+    this.#client.destroy();
   }
-  start(options) {
+
+  submit(worker, job_id, extranonce2, ntime, nonce) {
+    submitWork(worker, job_id, extranonce2, ntime, nonce, this.#client);
+  }
+
+  #start(options) {
     const updatedOptions = extend({}, defaultConfig, options);
 
     validateConfig(updatedOptions);
 
     const workObject = new WorkObject();
 
-    connect(client, updatedOptions);
+    this.#client = new net.Socket();
```

Comment:
Any specific reasons you want to instantiate here instead of module level?

Reply:
this is the purpose of the merge. Moving the socket as member of class Client allows you to instantiate and use multiple stratum clients in parallel. Having it at module level allows you to use only one Client.
In my application I open multiple connections to different stratum servers and I found a solution moving the socket from module level to private member